content/cloud-docs/agents/hooks.mdx

```diff
@@ -0,0 +1,157 @@
+
+- `post-apply` - This hook runs after `terraform apply` during an `apply`
+operation regardless whether or not `terraform apply` completed successfully.
+
+## Hook Behavior 
+
+Please note the following behavior when using hooks:
+
+- If a hook exits with a non-zero exit code, the Terraform run will fail
+immediately.
+- The standard output and standard error from the hook will be printed
+alongside the Terraform run output.
+- The name of each hook is not customizable.
+- Hooks must be executable. That is, each hook must have execute permission.
+- Each hook has a 60 second timeout.
+- There can only be one of each hook configured on an agent. That is, you can
+have both a `pre-plan` and `pre-apply` hook but you cannot have two `pre-plan`
+hooks.
+- Environment variables do not persist across hooks. For example, if a
+`pre-plan` hook exports environment variables, they will not be available
+during `terraform plan` nor will they be available during the `post-plan` hook.
+Similarly, if `terraform plan` exports environment variables, they will not be
+available during the `post-plan` hook.
+
+## Configuration
+
+Hooks are stored within the `hooks` subdirectory of the agent's data directory.
+By default, that's `~/.tfc-agent/hooks`.
+
+To configure a hook, create a file named `terraform-${HOOK}` where `${HOOK}`
+is one of the hooks listed above. Then store that file within the `hooks`
+subdirectory and make it executable. The agent will run your configured hooks
+as it processes Terraform runs.
+
+For example, a `pre-plan` hook is an executable file named
+`~/.tfc-agent/hooks/terraform-pre-plan`. When this hook is configured, an agent
+will run this hook before `terraform init` during a `plan` operation.
+
+### Running an Agent as a Binary
+
+Here's how you can configure hooks when running the `tfc-agent` binary
+directly on a Linux instance.
+
+1. Create the `hooks` directory within the agent's data directory.
+
+  ```sh
+  mkdir -p ~/.tfc-agent/hooks
+  ```
+
+1. Create your hooks. The following example creates a `pre-plan` hook.
+
+  ```sh
+  cat <<EOF > ~/.tfc-agent/hooks/terraform-pre-plan
+  #!/bin/bash
+  echo "Example pre-plan hook."
+  EOF
+  ```
+
+1. Make the hooks executable.
+
+  ```sh
+  chmod +x ~/.tfc-agent/hooks/terraform-pre-plan
+  ```
+
+1. Start the agent.
+
+  ```sh
+  tfc-agent
+  ```
+
+1. As the agent processes Terraform runs, it will run the hooks you've
+configured above.
+
+### Running an Agent with Docker 
+
+Configuring hooks when running `tfc-agent` using Docker requires you to build
+a new Docker image containing your hooks.
+
+1. Create a new directory and change into it. This limits the Docker build
+context to this new directory.
+
+  ```sh
+  mkdir hooks-docker-example
+  cd hooks-docker-example
+  ```
+
+1. Create a `hooks` directory to hold your hooks.
+
+  ```sh
+  mkdir hooks
+  ```
+
+1. Create your hooks. The following example creates a `pre-plan` hook.
+
+  ```sh
+  cat <<EOF > hooks/terraform-pre-plan
```

Comment:
can you add the $ in appropriate spots (lol I'm not 100% sure if any of this is output...*sigh* I need to get better at command line)

Reply:
I touched on this in this comment: https://github.com/hashicorp/terraform-website/pull/2154#discussion_r815242329

We can revisit adding `$` at a later time though. I'd rather keep better copy functionality rather than add the `$`.

For the commands on this page, they are meant to be copied in their entirety. There's no output from commands being shown, only the command itself to run.